src/analysis/retail/evoker/shared/modules/talents/TimeSpiral.tsx

```diff
@@ -0,0 +1,173 @@
+      return;
+    }
+    if (event.targetID === this.selectedCombatant.id) {
+      this.personalBuffsApplied += 1;
+    } else {
+      this.externalBuffsApplied += 1;
+    }
+    this.timeSpiralApplyTimestamps[event.targetID] = event.timestamp;
+    this.timeSpiralTimestampExists[event.targetID] = true;
+    this.timeSpiralDuration[event.targetID] = this.calculateTimeSpiralBuffDuration();
+  }
+
+  onRemoveBuff(event: RemoveBuffEvent) {
+    if (event.targetID === this.selectedCombatant.id) {
+      if (hasTimeSpiralConsumeEvent(event)) {
+        this.personalBuffsUsed += 1;
+      }
+    } else {
+      if (!this.timeSpiralTimestampExists[event.targetID]) {
+        //Can occur if Time Spiral was precast, but we have no way of knowing the buff duration in this case.
+        return;
+      }
+      // 900 (0.9 * 1000) is used to account for variations in timestamp.
+      if (
+        event.timestamp <
+        this.timeSpiralApplyTimestamps[event.targetID] +
+          this.timeSpiralDuration[event.targetID] * 900
+      ) {
+        // This can also be triggered by the player dying or cancelling the buff, but the former would require querying to check, and the latter is unlikely.
+        this.externalBuffsUsed += 1;
+      }
+    }
+    this.timeSpiralTimestampExists[event.targetID] = false;
+  }
+
+  calculateTimeSpiralBuffDuration() {
+    if (this.selectedCombatant.spec !== SPECS.AUGMENTATION_EVOKER) {
+      return TIME_SPIRAL_BASE_DURATION;
+    } else {
+      return (
+        TIME_SPIRAL_BASE_DURATION *
+        (1 +
+          TIMEWALKER_BASE_EXTENSION +
+          this.stats.currentMasteryPercentage * TIMEWALKER_EXTENSION_MULTIPLIER)
+      );
+    }
+  }
+
+  statistic() {
+    const externalBuffsWasted = Math.max(this.externalBuffsApplied - this.externalBuffsUsed, 0);
+    const personalBuffsWasted = Math.max(this.personalBuffsApplied - this.personalBuffsUsed, 0);
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.OPTIONAL(13)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={
+          <>Other players dying with or cancelling the buff will also trigger the 'used' count.</>
+        }
+      >
+        <TalentSpellText talent={TALENTS_EVOKER.TIME_SPIRAL_TALENT}>
+          <div>
+            <SoupIcon /> {this.personalBuffsUsed}
+            <small>
+              {' '}
+              personal <SpellLink spell={SPELLS.HOVER} /> casts gained
+            </small>
+            <br />
+            <WarningIcon /> {personalBuffsWasted}
+            <small>
+              {' '}
+              personal <SpellLink spell={SPELLS.HOVER} /> casts wasted
+            </small>
+            <br />
+            <InformationIcon /> {this.externalBuffsUsed}
+            <small> external buffs used</small>
+            <br />
+            <WarningIcon /> {externalBuffsWasted}
+            <small> external buffs unused</small>
+          </div>
+        </TalentSpellText>
```

Comment:
The wasted buffs should only be shown if relevant.
```suggestion
        <TalentSpellText talent={TALENTS_EVOKER.TIME_SPIRAL_TALENT}>
          <div>
            <SoupIcon /> {this.personalBuffsUsed}
            <small>
              {' '}
              personal <SpellLink spell={SPELLS.HOVER} /> casts gained
            </small>

            {personalBuffsWasted > 0 && (
              <>
                <br />
                <WarningIcon /> {personalBuffsWasted}
                <small>
                  {' '}
                  personal <SpellLink spell={SPELLS.HOVER} /> casts wasted
                </small>
              </>
            )}

            <br />
            <InformationIcon /> {this.externalBuffsUsed}
            <small> external buffs used</small>
            
            {personalBuffsWasted > 0 && (
              <>
                <br />
                <WarningIcon /> {externalBuffsWasted}
                <small> external buffs unused</small>
              </>
            )}
          </div>
        </TalentSpellText>
```